java/vector/src/main/codegen/templates/UnionVector.java

```diff
@@ -157,15 +157,16 @@ public void loadFieldBuffers(ArrowFieldNode fieldNode, List<ArrowBuf> ownBuffers
   @Override
   public List<ArrowBuf> getFieldBuffers() {
     List<ArrowBuf> result = new ArrayList<>(1);
-    setReaderAndWriterIndex();
-    result.add(typeBuffer);
+    result.add(sliceTypeBuffer());
```

Comment:
calling getFieldBuffers should not create new arrow buf objects. If we're relying on reader/writer index on this we should stop.